src/librustdoc/html/format.rs

```diff
@@ -62,6 +62,12 @@ crate struct Buffer {
     buffer: String,
 }
 
+impl core::fmt::Write for Buffer {
```

Comment:
I would probably just forward all three methods directly, since it's pretty trivial to do, and also annotate each of them with `#[inline]`.

And we should probably remove the inherent `write_str()` and `write_fmt()` methods, replacing their usage with these?

Reply:
I forwarded all the methods and then tried to use the trait methods but i could not make head or tail of the errors that came :(
I then tried to call the traits as `<Self as Write>::write_fmt(....)` which worked. Have not pushed those changes up though, as I am not sure if they are sane :(